apps/studio/components/interfaces/DataWarehouse/WarehouseCollectionDetail.tsx

```diff
@@ -0,0 +1,171 @@
+  const [params, setParams] = React.useState({
+    sql: `select current_timestamp() as 'time'`,
+  })
+
+  const [pagination, setPagination] = React.useState({
+    limit: 100,
+    offset: 0,
+  })
+
+  useEffect(() => {
+    if (collection) {
+      setParams((prevParams) => ({
+        ...prevParams,
+        sql: `
+        select id, timestamp, event_message from \`${collection.name}\`
+        where timestamp > timestamp_sub(current_timestamp(), interval 7 day)
+        order by timestamp desc limit ${pagination.limit} offset ${pagination.offset}
+        `,
+      }))
+    }
+  }, [collection, pagination])
+
+  const {
+    isLoading: queryLoading,
+    data: queryData,
+    isError,
+    refetch,
+    isRefetching,
+  } = useWarehouseQueryQuery(
+    { ref: projectRef, sql: params.sql },
+    {
+      enabled: !!params.sql,
+    }
+  )
+
+  useEffect(() => {
+    if (isError) {
+      toast.error('Error loading collection data')
+    }
+  }, [isError])
+
+  const formatResults = (results: any) => {
+    if (!results || !results.length) {
+      return []
+    }
+
+    const r = results.map(({ timestamp, ...r }: any) => {
+      return {
+        timestamp: new Date(timestamp / 1000).toLocaleString(),
+        ...r,
+      }
+    })
+
+    return r
+  }
+
+  const results = formatResults(queryData?.data?.result)
+
+  function loadOlder() {
+    setPagination({ ...pagination, offset: pagination.offset + pagination.limit })
+  }
+
+  const isLoading = queryLoading || collectionsLoading || isRefetching
+
+  return (
+    <>
+      <div className="relative flex flex-col flex-grow h-full">
+        <ShimmerLine active={isLoading} />
+        <LoadingOpacity active={isLoading}>
+          <div className="flex flex-col w-full">
+            <div className="flex justify-between items-center pr-3">
+              <h2 className="p-3">{collection?.name}</h2>
+              <div className="flex items-center gap-2">
+                <Button asChild type={'text'}>
+                  <Link href={`/project/${projectRef}/logs/collections/access-tokens`}>
+                    Access tokens
+                  </Link>
+                </Button>
+
+                <TestCollectionDialog
+                  accessTokens={accessTokens.data?.data || []}
+                  collectionToken={collectionToken}
+                  projectRef={projectRef}
+                  collections={collections || []}
+                />
+              </div>
+            </div>
+            <LogTable
+              collectionName={collection?.name}
+              queryType="warehouse"
+              hasEditorValue={false}
+              projectRef={projectRef}
+              isLoading={isLoading}
+              data={results}
+              params={params}
+              error={isError ? 'Error loading data' : undefined}
+              maxHeight="calc(100vh - 139px)"
+              hideHeader={true}
+            />
+          </div>
+        </LoadingOpacity>
+
```

Comment:
there needs to be a UI error handler, e.g: https://github.com/supabase/supabase/blob/feat/warehouse/apps/studio/components/interfaces/Auth/RateLimits/RateLimits.tsx#L134